### ISSUE TITLE
Question on the results of TransMIL and ABMIL

### ISSUE THREAD

**Issue by wyhsleep** [dazhangyu123/ACMIL#2]
Hi, it‘s a nice work, thanks for your contribution. But in table 1, I have some question about the performance of TransMIL and ABMIL. In your work, the results of ABMIL is better with both two backbones. 
<img width="731" alt="image" src="https://github.com/dazhangyu123/ACMIL/assets/74030174/9fe4a540-d0c4-4fd6-ac2f-ceca2c7a14b5">
But in origional TransMIL paper, the results is the opposite one. So have you thought about the reason？Can this account for the different backbone？Since in TransMIL, they use res50 to extract features.
<img width="961" alt="image" src="https://github.com/dazhangyu123/ACMIL/assets/74030174/3f84623e-a03a-4a0c-99c0-22fc9d9ae27a">

**Comment by dazhangyu123** [comment]
I share a similar question and haven't fully unraveled it. My attempt to explain is as follows: While TransMIL makes strides in considering the correlation between instances, the use of two stacked attention layers introduces additional parameters, making TransMIL more susceptible to overfitting. In such cases, the quality of representation becomes pivotal for effective TransMIL utilization, and the better representation is more beneficial to TransMIL. Notably, three groups of results used the backbones of ResNet18 (our paper), ResNet50 (TransMIL paper), and SSL pretrained ViT (our paper), and this choice significantly impacted the final outcomes. 

Furthermore, it's crucial to note that the implementation details and the selection of hyperparameters can also influence performance. In response to this, I've updated the implementations of ABMIL, TransMIL, and other baselines. I hope this provides some clarity.

**Comment by wyhsleep** [comment]
Thanks for your answering, that's may be one possible reason. And I have another question about Max-pooling. Since its performance on SSL pretrained ViT even surpass ABMIL. I occur this phenomena during my experiment. Can this phenomenon be explained by the fact that the feature representations are already good enough that there is no need to utilize additional parameters to learn?

**Comment by dazhangyu123** [comment]
Certainly, I share your viewpoint. I have obtained similar results on the CAMELYON dataset with ViT SSL pretrained embeddings.

**Comment by MrPeterJin** [comment]
Also, can the authors open-source their splits for reproducing the results? With "randomly split" quoted from the original work, the results are not easy to reproduce.

**Comment by dazhangyu123** [comment]
Thank you for your suggestion, I have uploaded the split files with five seeds for CAMELYON dataset. please check the files in [splits/camelyon](https://github.com/dazhangyu123/ACMIL/tree/main/splits/camelyon)

**Comment by wyhsleep** [comment]
Hi, i have one more question about what's the learning rate for reproducing the results of ABMIL and TransMIL？

**Comment by dazhangyu123** [comment]
we set learning rate as 0.0001 for ViT-based features and learning rate as 0.0002 for ResNet-based features for all MIL models.

**Comment by wyhsleep** [comment]
Thank you for your reply, so I notice in the paper you mentioned that you trainde the model for 100 epochs, so what's the model you use for the testing, the one with the minimal loss or just the one in the last epoch?

**Comment by dazhangyu123** [comment]
We observed that the model with the minimal validation loss and the one from the last epoch exhibit similar performance across three datasets. Following established studies, we chose to use the model with minimal validation loss. Thank you.

**Comment by CodeLiSW** [comment]
> Thank you for your suggestion, I have uploaded the split files with five seeds for CAMELYON dataset. please check the files in [splits/camelyon](https://github.com/dazhangyu123/ACMIL/tree/main/splits/camelyon)

Hi, thanks for your great work! I have some questions about reproducing results about ACMIL for CAMELYON16 dataset. I followed your dowload link "ImageNet supervised ResNet18", and your split files. However, I got different results below:

for ACMIL with mask_drop=0.6
```shell
CUDA_VISIBLE_DEVICES=0 python Step3_WSI_classification_ACMIL.py --seed 4 --wandb_mode disabled --arch ga --n_token 5 --n_masked_patch 10 --mask_drop 0.6 --pretrain natural_supervised --config config/camelyon_natural_supervised_config.yml
``` 
![Image](https://github.com/user-attachments/assets/47489111-cd9c-4e81-8a90-706e9120088e)

for ACMIL with mask_drop=0.0
```shell
CUDA_VISIBLE_DEVICES=0 python Step3_WSI_classification_ACMIL.py --seed 4 --wandb_mode disabled --arch ga --n_token 5 --n_masked_patch 10 --mask_drop 0.0 --pretrain natural_supervised --config config/camelyon_natural_supervised_config.yml
```
![Image](https://github.com/user-attachments/assets/edb3131c-bb64-48a3-a083-3ff685249096)

for ABMIL
```shell
CUDA_VISIBLE_DEVICES=0 python Step3_WSI_classification_ACMIL.py --seed 4 --wandb_mode disabled --arch ga --n_token 1 --n_masked_patch 0 --mask_drop 0 --pretrain natural_supervised --config config/camelyon_natural_supervised_config.yml
```
![Image](https://github.com/user-attachments/assets/a19636b3-9953-40b5-a44c-d1fe3844577f)

It seems ABMIL performances better than ACMIL using camelyon16 (extracted by resnet18), do u have any ideas about this difference? Am I wrong with the commands?

**Comment by dazhangyu123** [comment]
As specified in the paper, we used M = 2, K = 10, and p = 0.6 for this dataset and natural-supervised setting.

![Image](https://github.com/user-attachments/assets/5ef82f2b-0195-4eda-9058-f35da5208979)
